docs/rag.md

```diff
@@ -0,0 +1,36 @@
+Querying with OpenAI: When a query is made, it could be processed by OpenAI's GPT-4 to understand the context better. For example, GPT-4 could convert a natural language query into a more structured form that Azure AI Search can understand.
+Search and Retrieval: Azure AI Search retrieves the most relevant results based on the processed query. The results are ranked according to relevance, providing a rich search experience.
+Post-processing with OpenAI: The search results can be further processed by OpenAI's GPT-4 for better presentation. For example, GPT-4 could generate a natural language summary of the search results.
+Monitoring and Maintenance: The performance and usage of the search service can be tracked using Azure Monitor and Azure Log Analytics. The search indices, data sources, and other resources can be managed using the Azure portal, REST APIs, or .NET SDK.
+
+## RAG pattern on Azure
+
+![RAG](./media/aoaiendtoend.png)
+
+1. Data Ingestion: Azure Cognitive Search supports numerous data connectors that allow you to ingest data from various sources. For instance, you can use Azure SQL Database, Azure Blob Storage, Azure Data Lake Store, or Azure Cosmos DB. It also supports connectors for popular SaaS platforms like SharePoint and Salesforce, and for on-premises or other cloud data sources through Azure Data Factory.
+2. Skills in Azure Cognitive Search: The built-in skills in Azure Cognitive Search enable the extraction of key phrases, identification of language, people, organizations, locations, and more from unstructured text. You can also create custom skills to meet specific needs. These skills are part of the enrichment phase where raw input data is transformed and enriched before it gets indexed.
+3. Index Creation: Creating an index in Azure Cognitive Search is fundamental for efficient search operations. The schema defines fields in your data, the data types of these fields, and how these fields should be treated in search operations. The data source points to the location of your data. The indexer crawls over the data source and automatically extracts searchable data and metadata based on the predefined schema.
+4. Query Processing: Azure Cognitive Search supports full-text search queries, filters, facets, and even complex types. It uses a simple query syntax by default, but also supports the Lucene query syntax for more complex search queries.
+5. Azure Function as Application Orchestrator: Azure Functions can be used to orchestrate the flow of data, for example, sending prompts to Azure OpenAI and handling the responses. Azure Functions are serverless and event-driven, meaning they scale automatically to handle incoming traffic.
```

Comment:
Would it be better to say: 5. Application Orchestrator (for example, using Azure Functions)